### PR TITLE
CORTX-29708: Add proper validation of args in mock_health_event_publisher.py file

### PR DESCRIPTION
**Pull request by gargpalak048**
Signed-off-by: Palak Garg <palak.garg@seagate.com>

# Problem Statement
- Invalid argument validations are not properly handled.

# Design
-  Added validation to check for Invalid Argument

# Coding
-  [x] Coding conventions are followed and code is consistent

# Testing 
- [x] Unit and System Tests are added
https://jts.seagate.com/secure/attachment/514154/test_results_29708.txt
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM

# Review Checklist 
- [x] PR is self reviewed
- [x] JIRA number/GitHub Issue added to PR
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained
- [ ] Is there a change in filename/package/module or signature? [Y/N]: 
- [ ] If yes for above point, is a notification sent to all other cortx components? [Y/N]
- [ ] Side effects on other features (deployment/upgrade)? [Y/N]
- [ ] Dependencies on other component(s)? [Y/N]
-     If yes for above point, post link to the corresponding PR.

# Review Checklist 
- [ ] Is perfline test run and the report with and without the changes updated in the PR? [Y/N]: 

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Review by mukhtar-inamdar — APPROVED** [review]
just  give a single space before options in below
 print(f"Error: Invalid Argument:{option}\n"
other chnages looks good to me

**Comment by gargpalak048** [comment]
> just give a single space before options in below print(f"Error: Invalid Argument:{option}\n" other chnages looks good to me

Done

**Review by shaileshsvc — APPROVED** [review]
Approved

**Review by gauri-bhosale — APPROVED** [review]

**Comment by Madhura-08** [comment]
@gargpalak048 Please add Happy Path testing details. ie script execution with supported args.

**Review by akash2144 — APPROVED** [review]

**Comment by gargpalak048** [comment]
@Madhura-08 
Added Happy Test path details in test_results.txt file

**Review by Madhura-08 — APPROVED** [review]

File: ha/util/health_generator/mock_health_event_publisher.py [hunk 1]
```diff
@@ -205,9 +206,14 @@ def get_args() -> (argparse.Namespace, argparse.ArgumentParser):
...
 
     data_node_ids = get_data_nodes()
+    options_regex = re.compile('(-gdt$)|(-gs$)')
 
     if len(sys.argv) > 1:
         option = sys.argv[1]
+        if not options_regex.match(option):
+            print(f"Error: Invalid Argument: {option}\n"
+                  f"Please check supported args list: ['-gdt', '-gs']")
+            sys.exit(1)
```
Comment (mariyappanp) [inline ha/util/health_generator/mock_health_event_publisher.py:216]: args validation can be kept in get_args().   `args, parser_obj = get_args()`. 
Also, alternative args also should be considered for validation.
'-gdt', '--get-data-nodes'
'-gs', '--get-server-nodes'

Comment (gargpalak048) [reply ha/util/health_generator/mock_health_event_publisher.py]: @mariyappanp in my opinion, get_args() func should be only for parsing command line arguments.
Validation should be done outside.
If you check #L211, the len of args validation is also kept outside func only, aligning to that we can extend there only.
Added
 '-gdt', '--get-data-nodes'
'-gs', '--get-server-nodes'
args as well.

Comment (mariyappanp) [reply ha/util/health_generator/mock_health_event_publisher.py]: Okay. Leaving to author as my point is just a suggestion only here.

**Review by mariyappanp — APPROVED** [review]

**Comment by ArchanaLimaye** [comment]
@gargpalak048  please add link to test file in "Testing" section of the PR

**Review by ArchanaLimaye — APPROVED** [review]

**Comment by gargpalak048** [comment]
> @gargpalak048 please add link to test file in "Testing" section of the PR

Added

**Pull request merged**